src/test/kotlin/org/opensearch/replication/bwc/BackwardsCompatibilityIT.kt

```diff
@@ -139,12 +143,40 @@ class BackwardsCompatibilityIT : MultiClusterRestTestCase() {
                 Assertions.assertThat(getResponse.isExists).isTrue()
                 Assertions.assertThat(getResponse.sourceAsMap).isEqualTo(source)
             }, 60, TimeUnit.SECONDS)
+
+            //Check for latest retention lease when full cluster restart is done
+            if (ClusterStatus.from(System.getProperty("tests.bwcTask")) == ClusterStatus.FULL_CLUSTER_RESTART || ClusterStatus.from(
+                    System.getProperty("tests.bwcTask")) == ClusterStatus.ROLLING_UPGRADED) {
+                validateNewRetentionLeaseId(follower, leader)
```

Comment:
This might fail with rolling upgrade as we don't know when ShardReplicationTask restarts(with 1st/2nd/3rd node upgrade). 
Probably we can just rely on full cluster restart.